src/core/atomic.d

```diff
@@ -1273,6 +1182,146 @@ else version( AsmX86_64 )
     }
 }
 
+/* ***
+ * Functions built in to the compiler, along with a default implementation.
+ */
+
+bool cmpxchg(shared(ubyte)* here, ubyte ifThis, ubyte writeThis) pure nothrow @nogc @trusted
```

Comment:
> I'll just reiterate that D being a systems programming language, it should provide a means for the user to get access to low level things like intrinsics.

If that would be your goal then please build a proper API for the intrinsics that is compatible with other compilers, i.e. use https://gcc.gnu.org/onlinedocs/gcc-4.9.3/gcc/_005f_005fatomic-Builtins.html, so users don't have to deal with different intrinsics on every compiler.
Failing that let's please just make core.atomic faster and more useful.